internal/data/metric.go

```diff
@@ -599,33 +639,36 @@ type SummaryValueAtPercentile struct {
 	orig *otlpmetric.SummaryDataPoint_ValueAtPercentile
 }
 
-func NewSummaryValueAtPercentile() *SummaryValueAtPercentile {
-	return &SummaryValueAtPercentile{orig: &otlpmetric.SummaryDataPoint_ValueAtPercentile{}}
+// NewSummaryValueAtPercentile creates a new SummaryValueAtPercentile.
+func NewSummaryValueAtPercentile() SummaryValueAtPercentile {
+	return SummaryValueAtPercentile{orig: &otlpmetric.SummaryDataPoint_ValueAtPercentile{}}
 }
 
 // NewSummaryValueAtPercentileSlice creates a slice of SummaryValueAtPercentile that are correctly initialized.
 func NewSummaryValueAtPercentileSlice(len int) []SummaryValueAtPercentile {
+	// Slice for underlying data.
 	origs := make([]otlpmetric.SummaryDataPoint_ValueAtPercentile, len)
+	// Slice for wrappers.
 	wrappers := make([]SummaryValueAtPercentile, len)
 	for i := range origs {
 		wrappers[i].orig = &origs[i]
 	}
 	return wrappers
 }
 
-func (vp *SummaryValueAtPercentile) Percentile() float64 {
+func (vp SummaryValueAtPercentile) Percentile() float64 {
 	return vp.orig.Percentile
 }
 
-func (vp *SummaryValueAtPercentile) SetPercentile(v float64) {
+func (vp SummaryValueAtPercentile) SetPercentile(v float64) {
```

Comment:
Without pointer receiver set is not meaningful.

Reply:
it is because of pimpl

Reply:
sorry, my bad. it is not an issue.